### PR TITLE
vmm: Add a constant SGX_PAGE_SIZE variable in memory_manager.rs

### PR DESCRIPTION
**Pull request by optimistyzy**
Purpose: Do not directly use 0x1000 but use predefined constant value.

Signed-off-by: Ziye Yang <ziye.yang@intel.com>

### CODE REVIEW

File: vmm/src/memory_manager.rs [hunk 1]
```diff
@@ -56,6 +56,9 @@ const SNAPSHOT_FILENAME: &str = "memory-ranges";
 #[cfg(target_arch = "x86_64")]
 const X86_64_IRQ_BASE: u32 = 5;
 
+#[cfg(target_arch = "x86_64")]
+const SGX_PAGE_SIZE_SHIFT: u64 = 12;
```
Comment (sboeuf) [inline vmm/src/memory_manager.rs:60]: You could go one step further and declare another constant:
```rust
#[cfg(target_arch = "x86_64")]
const SGX_PAGE_SIZE: u64 = 1 << SGX_PAGE_SIZE_SHIFT;
```

File: vmm/src/memory_manager.rs [hunk 2]
```diff
@@ -1613,20 +1616,24 @@ impl MemoryManager {
...
 
             epc_region_size += epc_section.size;
         }
 
         // Place the SGX EPC region on a 4k boundary between the RAM and the device area
-        let epc_region_start =
-            GuestAddress(((self.start_of_device_area.0 + 0xfff) / 0x1000) * 0x1000);
+        let epc_region_start = GuestAddress(
+            ((self.start_of_device_area.0 + sgx_page_size - 1) >> SGX_PAGE_SIZE_SHIFT)
+                << SGX_PAGE_SIZE_SHIFT,
```
Comment (rbradford) [inline vmm/src/memory_manager.rs:1634]: The use of division is clearer for the page boundary calculation. I don't think the use of the constant for the page size adds anything.

File: vmm/src/memory_manager.rs (hunk 1 above)
Comment (optimistyzy) [reply vmm/src/memory_manager.rs:60]: sure

File: vmm/src/memory_manager.rs (hunk 2 above)
Comment (optimistyzy) [reply vmm/src/memory_manager.rs:1634]: Hi @rbradford Usually,  >> is faster than using /.  Though 4KB is a static size defined by SGX, but if we use a constant size, which will be better if SGX supports other size later.

Comment (rbradford) [reply vmm/src/memory_manager.rs:1636]: The rust compiler knows that division or multiplication by a constant multiple of 2 can be turned into shifts. If you want to extract the page size into a constant then that is fine, but I don't approve of the unnecessary change to use a shift and the extra constant required.

**Review by rbradford — CHANGES_REQUESTED** [review]

File: vmm/src/memory_manager.rs (hunk 2 above)
Comment (optimistyzy) [reply vmm/src/memory_manager.rs:1636]: @rbradford Let me remove the << >> operations. Thanks.

**Comment by optimistyzy** [comment]
> 

Fixed. Thanks.

**Review by rbradford — APPROVED** [review]

**Pull request merged**